src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScorer.java

```diff
@@ -373,6 +371,36 @@ public static boolean isValidAlignment(final Cigar cigar, final int numMismatche
     }
 
     /**
+     * Adds each node's score to its (super)kingdom's total in normalizationSums and assigns the node's PSPathogenTaxonScore
+     *  kingdom in taxIdsToScores. If the node does not have a kingdom, it is assigned to the root by default, which
+     *  is tallied as its own kingdom.
+     */
+    private static void assignKingdoms(final Map<String, PSPathogenTaxonScore> taxIdsToScores,
+                                      final Map<String, Double> normalizationSums,
+                                      final PSTree tree, final boolean normalizeByKingdom) {
+        for (final Map.Entry<String, PSPathogenTaxonScore> entry : taxIdsToScores.entrySet()) {
+            final String taxonId = entry.getKey();
+            final PSPathogenTaxonScore score = entry.getValue();
+            if (normalizeByKingdom) {
+                final List<String> path = tree.getPathOf(taxonId);
+                for (final String nodeId : path) {
+                    if (tree.getRankOf(nodeId).equals(KINGDOM_RANK_NAME) || tree.getRankOf(nodeId).equals(SUPERKINGDOM_RANK_NAME)
+                            || nodeId.equals(PSBuildReferenceTaxonomyUtils.ROOT_ID)) {
+                        normalizationSums.putIfAbsent(nodeId, 0.);
+                        normalizationSums.put(nodeId, normalizationSums.get(nodeId) + score.selfScore);
```

Comment:
makes more sense (efficient) get rid of putIfAbsent and then use ```getOrDefault(nodeId, 0.)``` below.

Reply:
Done